File: ocaml/3096.ml
Scanf.scanf "%i" (fun n -> 
    let r = floor (float n *. log10 (float n /. exp 1.0) +. log10 (2.0 *. 3.14159265358979312 *. float n) /. 2.0) +. 1.0 in 

    Printf.printf "%d\n" (int_of_float r)
  )

File: c#/3096.cs
using System; 
class URI {
    static void Main(string[] args) { 
        int n = Int32.Parse(Console.ReadLine());
        
        int r = (int) Math.Floor(n*Math.Log10(n/Math.E) + Math.Log10(2*Math.PI*n)/2.0) + 1;
        Console.WriteLine(r);
    }
}